docs/source/migration.mdx

```diff
@@ -445,117 +579,73 @@ const apolloServerInstance = new ApolloServer<MyContext>({
+### `requestAgent` option to `ApolloServerPluginUsageReporting`
 
-The `expressMiddleware` function accepts an `ApolloServer` instance, and as its name implies, you can set up `expressMiddleware` while adding other middleware to your `Express` server:
+The usage reporting plugin allows you to entirely replace its HTTP client via the `fetcher` option. Additionally, in Apollo Server 3, you could use an older `requestAgent` option, which would be passed to the `fetcher` function via the non-standard `agent` option.
 
-<MultiCodeBlock>
+Apollo Server 4 removes the `requestAgent` operation from `ApolloServerPluginUsageReporting`, and now only passes options that are part of the Fetch API spec to its `fetcher`. If you used `requestAgent` before, you can use the `node-fetch` npm package to override `fetcher` yourself. So if you used `requestAgent` like this:
 
-```ts {18}
-import express from 'express';
-import http from 'http';
-import cors from 'cors';
-import { json } from 'body-parser';
-import { ApolloServer, ApolloServerPluginDrainHttpServer } from '@apollo/server';
-import { expressMiddleware } from '@apollo/server/express4';
-import { typeDefs, resolvers } from ('./schema');
+```ts
+ApolloServerPluginUsageReporting({ requestAgent })
+```
 
-// Set up Express
-const app = express();
-const httpServer = http.createServer(app);
+You can now write:
 
-// Set up Apollo Server
-const server = new ApolloServer({
-  typeDefs,
-  resolvers,
-  plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
+```ts
+import fetch from 'node-fetch';
+ApolloServerPluginUsageReporting({
+  fetcher: (url, options) => fetch(url, {
+    ...options,
+    agent: requestAgent,
+  }),
 });
-await server.start();
-
-// Pass an ApolloServer instance to the expressMiddleware function
-app.use(cors(), json(), expressMiddleware(server));
-await new Promise<void>((resolve) => httpServer.listen({ port: 4000 }, resolve));
-
-console.log(`🚀 Server ready at http://localhost:4000`);
 ```
 
-```js {18}
-import express from 'express';
-import http from 'http';
-import cors from 'cors';
-import { json } from 'body-parser';
-import { ApolloServer, ApolloServerPluginDrainHttpServer } from '@apollo/server';
-import { expressMiddleware } from '@apollo/server/express4';
-import { typeDefs, resolvers } from ('./schema');
 
-// Set up Express
-const app = express();
-const httpServer = http.createServer(app);
+### Older Gateway types
 
-// Set up Apollo Server
-const server = new ApolloServer({
-  typeDefs,
-  resolvers,
-  plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
-});
-await server.start();
+The `gateway` option to the `ApolloServer` constructor is designed to be used with Apollo Gateway, but can also be used with your own implementation of the gateway interface. Several changes have been made to that interface in Apollo Server 4.
```

Comment:
I mean sorta but not really. The `gateway` interface is a kinda random hodgepodge of "let you use a different execution algorithm" and "let the schema change over time". Which is designed for Apollo Gateway but certainly
people have figured out that it's a way to run a server whose schema changes over time without it being federation.

Reply:
Yeah that's still not quite it.

We defined a particular API that lets you connect Apollo Gateway to AS if you want to be the gateway for a federated graph.

But what that API actually is is not a "federated graph gateway" API.  It's a "replace the execution function and/or allow the schema to change over time" API.  Plenty of people are happy to use this API because they find one or both of these capabilities interesting for reasons unrelated to federation.

This section is about changes we've made to the API that are irrelevant if you're using a recent version of Apollo Gateway (and a separate section mentions the new AG version requirement) — because we know these versions of Apollo Gateway mesh with the changes in question. But if you're using the API for unrelated reasons and writing your own implementation (not something we've super documented or super encouraged) then this is relevant.

Reply:
8ceed8c